internal/xds/xds.go

```diff
@@ -40,3 +42,24 @@ func GetXDSHandshakeClusterName(attr *attributes.Attributes) (string, bool) {
 	name, ok := v.(string)
 	return name, ok
 }
+
+// xdsHashKey is the type used as the key to store request hash in the context
+// used when combining the Ring Hash load balancing policy with xDS.
+type xdsHashKey struct{}
+
+// GetXDSRequestHash returns the request hash in the context and true if it was set
+// from the xDS config selector. If the xDS config selector has not set the hash,
+// it returns 0 and false.
+func GetXDSRequestHash(ctx context.Context) (uint64, bool) {
```

Comment:
For those two functions I decided to follow the convention established above with Get/SetXDSHandshakeClusterName despite the redundancy with the xds package name.

XDSHeaderName violates the rule that says that function names should not start with the package name, so I added the Get prefix, but I could also go with `xds.SetRequestHash` and `xds.RequestHash`. Let me know if you have a preference.